### PR TITLE
Improve cloud particles

### PR DESCRIPTION
**Pull request by bruhmoent**
This PR:
- Enables the change of X and Y speeds,
- Enhances the intensity setting 

Note:
This pr is waiting for simplesquirrel, as I'm unsure how is it handled; hence that's why I didn't regenerate the wrapper.

### CODE REVIEW

File: src/object/cloud_particle_system.cpp [hunk 1]
```diff
@@ -46,56 +48,104 @@ CloudParticleSystem::CloudParticleSystem(const ReaderMapping& reader) :
...
-      m_current_speed += (m_target_speed - m_current_speed) * amount;
-      m_speed_fade_time_remaining -= dt_sec;
+  // Update X speed.
+  if (m_speed_fade_time_remaining_x > 0.f)
+  {
+    if (dt_sec >= m_speed_fade_time_remaining_x)
+    {
+      m_current_speed_x = m_target_speed_x;
+      m_speed_fade_time_remaining_x = 0.f;
+    } else
```
Comment (tobbi) [inline src/object/cloud_particle_system.cpp:129]: Can you put the `else` in a new line please.

File: src/object/cloud_particle_system.cpp [hunk 2]
```diff
@@ -46,56 +48,104 @@ CloudParticleSystem::CloudParticleSystem(const ReaderMapping& reader) :
...
+  }
+
+  // Update Y speed.
+  if (m_speed_fade_time_remaining_y > 0.f)
+  {
+    if (dt_sec >= m_speed_fade_time_remaining_y)
+    {
+      m_current_speed_y = m_target_speed_y;
+      m_speed_fade_time_remaining_y = 0.f;
+    } else
```
Comment (tobbi) [inline src/object/cloud_particle_system.cpp:144]: Same here.

File: src/object/cloud_particle_system.cpp [hunk 3]
```diff
@@ -119,15 +171,18 @@ void CloudParticleSystem::update(float dt_sec)
...
+      if (dt_sec >= cloudParticle->target_time_remaining)
+      {
         cloudParticle->alpha = cloudParticle->target_alpha;
         cloudParticle->target_time_remaining = 0.f;
         if (cloudParticle->alpha == 0.f) {
           // Remove this particle, but not at this point
           // as it would interfere with the iterator.
         }
-      } else {
+      } else
```
Comment (tobbi) [inline src/object/cloud_particle_system.cpp:184]: Dito here.

File: src/object/cloud_particle_system.cpp [hunk 4]
```diff
@@ -138,15 +193,34 @@ void CloudParticleSystem::update(float dt_sec)
...
       particles.erase(particles.begin()+i);
   }
 }
 
-int CloudParticleSystem::add_clouds(int amount, float fade_time)
+void
+CloudParticleSystem::apply_fog_effect(DrawingContext& context)
+{
+  float opacity = fog_max_value * (m_current_amount - fog_start_amount) / (max_amount - fog_start_amount);
+  if (opacity < 0.f)
```
Comment (tobbi) [inline src/object/cloud_particle_system.cpp:209]: Maybe use `math::clamp`? @MatusGuy Isn't that right?

**Review by tobbi — CHANGES_REQUESTED** [review]
Please check the `else` placement for all the code. As for math::clamp, I'd like to ask for input from Marty.

File: src/object/cloud_particle_system.cpp (hunk 4 above)
Comment (MatusGuy) [reply src/object/cloud_particle_system.cpp:209]: Yeah

**Review by tobbi — CHANGES_REQUESTED** [review]
One more thing:
CI on Linux throws this error:
```
 /home/runner/work/supertux/supertux/src/object/cloud_particle_system.cpp:208:36: error: implicit conversion from 'int' to 'float' may lose precision [-Werror,-Wimplicit-int-float-conversion]
  float opacity = fog_max_value * (m_current_amount - fog_start_amount) / (max_amount - fog_start_amount);
```

File: src/object/cloud_particle_system.cpp [hunk 5]
```diff
@@ -138,15 +197,31 @@ void CloudParticleSystem::update(float dt_sec)
...
+CloudParticleSystem::apply_fog_effect(DrawingContext& context)
+{
+  float opacity = fog_max_value * (static_cast<float>(m_current_amount) - static_cast<float>(fog_start_amount)) / (static_cast<float>(max_amount) - static_cast<float>(fog_start_amount));
+  opacity = std::clamp(opacity, 0.f, 1.f);
+
+  context.push_transform();
+  context.set_translation(Vector(0, 0));
+  context.color().draw_filled_rect(context.get_rect(),
+    Color(0.3f, 0.38f, 0.4f, opacity),
+    199); // TODO: Change the hardcoded layer value with the cloud's layer
```
Comment (MatusGuy) [inline src/object/cloud_particle_system.cpp:219]: Fix this todo

File: src/scripting/clouds.cpp
Comment (MatusGuy) [inline src/scripting/clouds.cpp:1]: Revert your changes in files inside the scripting/ folder, as they'll be replaced in #2792.

**Review by MatusGuy — CHANGES_REQUESTED** [review]

**Comment by bruhmoent** [comment]
Don't worry about failing builds, I'm waiting for simplesquirrel.

**Pull request closed without merging**